sympy/solvers/bivariate.py

```diff
@@ -199,17 +199,40 @@ def _solve_lambert(f, symbol, gens):
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
+    even_degrees = [i for i in degree_list(lhs) if i%2 == 0]
+    t = Dummy('t', **symbol.assumptions0)
+    if len(even_degrees) != 0:
+        for i in even_degrees:
+            lhs = lhs.xreplace({symbol**i: t**i})
+
     lamcheck = [tmp for tmp in gens
                 if (tmp.func in [exp, log] or
                 (tmp.is_Pow and symbol in tmp.exp.free_symbols))]
     if not lamcheck:
         raise NotImplementedError()
 
+    if lhs.is_Add and lhs.has(t) and len(even_degrees) == 1:
```

Comment:
What if `len(even_degrees)` > 1? Any specific reason to have this check?

Reply:
Yes suppose there are more than one even degrees are present like `t**2 + t**4 + t_independent_term`. 
And this will add into the code so `diff = expand_log(log(t**2+t**4) - log(t_independent_term))`
which will not be expanded which is not required here and is useless to us.
```python
>>>t = Symbol('t', positive = True)
>>>expand_log(log(t**2 + t**4))
log(t**4 + t**2)
```

Reply:
Okay, then you should do the replacement (symbol -> t) only when `len == 1` above. This will avoid unnecessary re-replacement to the original equation below.
Also please verify if this goes in accordance with `Mul` types.

Reply:
Another thing, the `.is_Add` case was not present before. What caused this to be included? Which equations get here?

Reply:
> then you should do the replacement (symbol -> t) only when len == 1 above

No its only the case `.is_Add` when it is required. `Mul` doesn't have that type of requirement.

Reply:
> Another thing, the .is_Add case was not present before. What caused this to be included? Which equations get here?

Clearly from above discussion it was made to handle these type of equation `solve(x**2 - 2**x, x)` which were not including `-2*LambertW(log(2)/2)/log(2)` in the solution.

Reply:
> Also please verify if this goes in accordance with Mul types.
The logic of handling when `lhs.is_Add` and `lhs.is_Mul` are completely different in the way they are handled.
Moreover `lhs.is_Mul` can have any number of even_degrees because Mul type is easily converted to log form rather than the case with `lhs.is_Add`.

Reply:
Are you just trying to figure out if the expression is in the form `a*x**even + b`?

Reply:
No @smichr , `a*x**even + b` this is only a subset of equations you are thinking we are also considering here equations `a*x**even + b* f(containing odd powers of x)`. I was just checking if even degrees is of symbol is present or not and in caseof .is_Add I am only considering one even degree and the reason is mentioned in this [comment](https://github.com/sympy/sympy/pull/16890#discussion_r293578394).

Reply:
So in the expanded expression you are looking for a single term with an even power of the generator:
```python
exeq = eq.expand()
if exeq.is_Add:
    odd, even = sift(exeq, lambda t: any(
        i.is_Pow and i.base == x and i.exp % 2 == 0 for i in Mul.make_args(t)), binary=True)
    if len(even) == 1:
        ok
```

Reply:
The above would pass for a symbolic exponent, too. If you don't want that then you should use `i.exp.is_Integer and i.exp % 2 == 0`.

Reply:
@smichr I tried your code but it gives this error one way or another. 
```
/Gsoc/sympy/sympy/utilities/iterables.py in sift(seq, keyfunc, binary)
    699         return m
    700     sift = F, T = [], []
--> 701     for i in seq:
    702         try:
    703             sift[keyfunc(i)].append(i)

TypeError: 'Add' object is not iterable
```
> you are looking for a single term with an even power of the generator:

We are not looking for just a single term here. Sorry If it seems that way but what I meant was that all the expressions involving even degrees after applying all algorithms of log will be converted to logarithms expression which may have another form as described above. The expression here can be very complex also . So the above approach was applied after discussing with mentors @Yathartha22 @aktech  to replace all the even degrees with a symbol t and after all operations done change the even degrees to + symbol and -symbol . The expressions in which the even degrees were not changed to log will be remain same as -sym**even and sym**even will become sym**even eventually.

Reply:
@Yathartha22 @aktech what do you think about the code adviced by @smichr ?? I might be missing some understanding about it.